### PR TITLE
[FLINK-25549][flink-dstl] [JUnit5 Migration] Module: flink-dstl

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

[JUnit5 Migration] Module: flink-dstl

## Brief change log

[JUnit5 Migration] Module: flink-dstl

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @AHeise , could you help me to review this patch ? Thank you very much.

**Comment by gaoyunhaii** [comment]
Thanks @RocMarshal for the migrating! I'll have a look~

File: flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogWriterSqnTest.java [hunk 1]
```diff
@@ -72,14 +68,9 @@
...
-    private final WriterSqnTestSettings test;
-
-    public FsStateChangelogWriterSqnTest(WriterSqnTestSettings test) {
-        this.test = test;
-    }
-
-    @Test
-    public void runTest() throws IOException {
+    @MethodSource("getSettings")
+    @ParameterizedTest
```
Comment (gaoyunhaii) [inline flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogWriterSqnTest.java:72]: Might complement the display name with `@ParameterizedTest(name = "name = {0}")`

File: flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogWriterSqnTest.java [hunk 2]
```diff
@@ -91,24 +82,14 @@ public void runTest() throws IOException {
...
             }
             test.action.accept(writer);
-            assertEquals(
-                    getMessage(),
-                    test.expectIncrement
-                            ? writer.initialSequenceNumber().next()
-                            : writer.initialSequenceNumber(),
-                    writer.lastAppendedSqnUnsafe());
+            assertThat(writer.lastAppendedSqnUnsafe())
+                    .isEqualTo(
```
Comment (gaoyunhaii) [inline flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogWriterSqnTest.java:86]: Might keeps the customized error message with 
```
assertThat(...).as(test.getMessage()).isEqualTo(...)
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/changelog/inmemory/StateChangelogStorageTest.java [hunk 3]
```diff
@@ -44,31 +45,43 @@
...
+    public static Stream<Boolean> parameters() {
+        return Stream.of(true);
     }
 
-    @Test
-    public void testWriteAndRead() throws Exception {
+    @MethodSource("parameters")
+    @ParameterizedTest
+    public void testNoAppendAfterClose(boolean compression) throws IOException {
+        assertThatThrownBy(
```
Comment (gaoyunhaii) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/changelog/inmemory/StateChangelogStorageTest.java:65]: Narrow down the scope of `assertThatThrownBy` if possible?

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/changelog/inmemory/StateChangelogStorageTest.java [hunk 4]
```diff
@@ -44,31 +45,43 @@
...
+                                                    new OperatorID().toString(),
+                                                    KeyGroupRange.of(0, 0));
+                            writer.close();
+                            writer.append(0, new byte[0]);
+                        })
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    @MethodSource("parameters")
+    @ParameterizedTest
```
Comment (gaoyunhaii) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/changelog/inmemory/StateChangelogStorageTest.java:79]: Keeps the original description like `@ParameterizedTest(name = ...)` ?

**Review by gaoyunhaii — COMMENTED** [review]
Thanks @RocMarshal for the PR! I have left some small comments~

File: flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogStorageTest.java [hunk 5]
```diff
@@ -21,27 +21,24 @@
...
-
+import java.io.File;
 import java.io.IOException;
+import java.util.stream.Stream;
 
 import static org.apache.flink.changelog.fs.UnregisteredChangelogStorageMetricGroup.createUnregisteredChangelogStorageMetricGroup;
 
 /** {@link FsStateChangelogStorage} test. */
-@RunWith(Parameterized.class)
 public class FsStateChangelogStorageTest extends StateChangelogStorageTest {
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogStorageTest.java]: Good catch

File: flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploadSchedulerTest.java [hunk 6]
```diff
@@ -128,20 +127,18 @@ public void testDelay() throws Exception {
...
-                    assertTrue(probe.getUploaded().isEmpty());
-                    assertTrue(
-                            scheduler.getAllNonPeriodicScheduledTask().stream()
-                                    .anyMatch(
-                                            scheduled ->
-                                                    scheduled.getDelay(MILLISECONDS) == delayMs));
+                    assertThat(probe.getUploaded()).isEmpty();
+                    assertThat(scheduler.getAllNonPeriodicScheduledTask())
+                            .anyMatch(scheduled -> scheduled.getDelay(MILLISECONDS) == delayMs)
+                            .hasSizeGreaterThan(0);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploadSchedulerTest.java]: @gaoyunhaii Nice catch.

**Review by gaoyunhaii — APPROVED** [review]
Thanks @RocMarshal for the update! LGTM

**Pull request closed without merging**